woocommerce/woocommerce-gutenberg-products-block
# Fix/cart button focus hover states

This PR is intended to fix this issue #5647. The cart and mini-cart buttons didn't have hover & focus styles.

Our WC Block Button component is based on [Gutenberg's button](https://wordpress.github.io/gutenberg/?path=/story/components-button--buttons). The package name is [@wordpress/components](https://www.npmjs.com/package/@wordpress/components). This package's latest version displays a correct hover & focus style on the button component, but we are using an old version. We faced some issues with the bundle size of the newer versions because they include some libraries we don't need.

So, the solution is to set our focus and hover style for now. 

The latest Gutenberg's Button component introduced the `variant` attribute for "text", "contained" and "outlined" buttons. And since we are using an older version of this component that doesn't have this attribute, and we are already using the "contained" and "outlined" buttons in our plugin, implementing it on our end seems to be a good idea.

### Screenshots

Default state (filled button)
:-------------------------:
![image](https://user-images.githubusercontent.com/14235870/152257987-ea5968ae-6dbc-4f8d-b037-b7302ec9fe7f.png) 

Hover state (filled button)
:-------------------------:
![image](https://user-images.githubusercontent.com/14235870/152258416-e2c646e8-8d9f-4565-b70a-fa1043127ea7.png)

Focus state (filled button)
:-------------------------:
![image](https://user-images.githubusercontent.com/14235870/152258490-39c011e0-b6cc-4cf0-9a89-142772f6a681.png)

Default state (outlined button)
:-------------------------:
![image](https://user-images.githubusercontent.com/14235870/152258555-50438e6f-fdc9-4a7e-b62e-dbe00ab3bd3a.png)

Hover state (outlined button)
:-------------------------:
![image](https://user-images.githubusercontent.com/14235870/152258608-b944142a-a60f-4153-8190-e3412f58f4df.png)

Focus state (outlined button)
:-------------------------:
![image](https://user-images.githubusercontent.com/14235870/152258652-a38349b2-55a4-4718-91a5-f90d9ba19374.png)

### TODOS


- [X] Introduce the `outlined button` and replace the mini-cart custom styles

### Testing

### Manual Testing

1. Checkout the branch
2. Create test pages for cart, mini cart, and checkout blocks.
3. In each page check the hover and focus styles for the `View my cart`, `Got to checkout`, `Proceed to Checkout`, and `Place Order` buttons.
4. You can do the same test using the available default themes and the Strofront theme as well.
### Changelog

> Fixed hover and focus states for button components
